File: 0014 Ceasar Cypher/CeasarCypher.py
# O(n) Time and space complexity

def caesarCipherEncryptor(string, key):
	newLetters = []
	newKey = key % 26
    alphabet = list("abcdefghijklmnopqrstuvwxyz")
	for letter in string:
		newLetters.append(getNewLetter(letter, newKey, alphabet))
	return "".join(newLetters)
	
def getNewLetter(letter,key,alphabet):
	newLetterCode = alphabet.index(letter) + key
	return alphabet[newLetterCode%26]

### simplified

def caesarCipherEncryptor(string, key):
    newLetters = []
	newKey = key % 26
	for letter in string:
		newLetters.append(getNewLetter(letter,newKey))
	return "".join(newLetters)
    
def getNewLetter(letter, key):
	newLetterCode = ord(letter) + key
	return chr(newLetterCode) if newLetterCode <= 122 else chr(96 + newLetterCode % 122)

File: 014 Ceasar Cypher/CeasarCypher.js
// O(n) Time and space complexity

function caesarCipherEncryptor(string, key) {
  const newLetter = []
	const alphabet = "abcdefghijklmnopqrstuvwxyz".split('');
	const newKey = key % 26;
	for (const letter of string){
		newLetter.push(getNewLetter(letter, newKey, alphabet))
	}
	return newLetter.join('');
}
function getNewLetter(letter,key, alphabet){
	const newIndex = alphabet.indexOf(letter) + key;
	return alphabet[newIndex % 26];
}


// simplified

function caesarCipherEncryptor(string, key) {
	const newLetters = [];
	const newKey = key % 26;
	for (const letter of string){
		newLetters.push(getNewLetter(letter, newKey));
	}
  return newLetters.join('');
}
function getNewLetter(letter,key){
	const newLetterCode = letter.charCodeAt(letter)+key;
	return newLetterCode <=122 ? String.fromCharCode(newLetterCode) : String.fromCharCode(96 + (newLetterCode % 122)) 
}